### PR TITLE
add order-add-to-cart-api and unit test

### PR DESCRIPTION
**Pull request by SoniaChoo**

### CODE REVIEW

File: db/orders.go [hunk 1]
```diff
@@ -127,3 +127,110 @@ func UpdateTotalPriceInOrder(total_price float64, order_id int) error {
...
+	//if len(carts) = 0, insert to order_detail, else, update dish number according dish_id
+	if len(carts) == 0 {
+		if err = InsertDishToCart(detail); err != nil {
+			log.Printf("record insert into table order_detail with error, error is %s\n", err.Error())
+			return err
+		}
+		return nil
+	}
+	//brfore upadte dish number, we need to get the original number of this dish in cart
+	detail.Number += carts[0].Number
```
Comment (ziyitony) [inline db/orders.go:236]: please add condition to judge if length is 1.

File: db/orders.go [hunk 2]
```diff
@@ -127,3 +127,110 @@ func UpdateTotalPriceInOrder(total_price float64, order_id int) error {
...
+	db, err := DBFactory()
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return err
+	}
+
+	//start to excute SQL query
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	_, err = db.QueryContext(ctx, "insert into order_detail(detail_id, rest_id, order_id, dish_id, price, number, status) Values(?,?,?,?,?,?,?)", detail.DetailId, detail.RestId, detail.OrderId, detail.DishId, detail.Price, detail.Number, InCartStatus)
```
Comment (ziyitony) [inline db/orders.go]: you don't have detail_id here, please update your sql

File: db/orders.go [hunk 3]
```diff
@@ -127,3 +127,110 @@ func UpdateTotalPriceInOrder(total_price float64, order_id int) error {
...
+	db, err := DBFactory()
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return err
+	}
+
+	//start to excute SQL query
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	if _, err = db.QueryContext(ctx, "update order_detail set number = ? where dish_id = ?", detail.Number, detail.DishId); err != nil {
```
Comment (ziyitony) [inline db/orders.go]: please try to use PK if you have it.

**Review by ziyitony — CHANGES_REQUESTED** [review]

File: db/orders.go [hunk 4]
```diff
@@ -127,3 +127,110 @@ func UpdateTotalPriceInOrder(total_price float64, order_id int) error {
...
+	}
+	//brfore upadte dish number, we need to get the original number of this dish in cart
+	detail.Number += carts[0].Number
+	if err = UpdateDishToCart(detail); err != nil {
+		log.Printf("record update dish number in table order_detail with error, error is %s\n", err.Error())
+		return err
+	}
+
+	//update dish favorite when add dish to cart
+	if err = UpdateDishFavorite(detail.Number, detail.DishId); err != nil {
```
Comment (ziyitony) [inline db/orders.go]: the favorite count is not the same as detail.Number

File: db/orders.go (hunk 1 above)
Comment (SoniaChoo) [reply db/orders.go:236]: updated.

File: db/orders.go (hunk 2 above)
Comment (SoniaChoo) [reply db/orders.go]: updated.

File: db/orders.go (hunk 3 above)
Comment (SoniaChoo) [reply db/orders.go]: updated.

File: db/orders.go (hunk 4 above)
Comment (SoniaChoo) [reply db/orders.go]: updated.

**Pull request merged**